ports/nrf/boards/microbit/modules/boardmodules.mk

```diff
@@ -1,7 +1,6 @@
 BOARD_MICROBIT_DIR = boards/microbit/modules
 
 INC += -I./$(BOARD_MICROBIT_DIR)
-CFLAGS += -DBOARD_SPECIFIC_MODULES
```

Comment:
I think this still needs to be defined... the microbit module defines `board_modules_init0()` which still needs to be called from main.

Actually it's a bit confusing what should happen here.  There are two `boardmodules.h` files (one deleted by this PR, another in the pca10059 board dir).

Reply:
Ah I missed the dual-purpose here. I have done the simplest change, we should revisit board_init in the future and make it work consistently with the other ports.